### ISSUE TITLE
Cannot set custom headers with $fetch

### ISSUE THREAD

**Issue by RijaCloud** [nuxt/framework#7722]
### Environment

Operating System: Windows 10
Node Version: v14.16.1
Nuxt Version: 3.0.0-rc.08
Builder: vite
Runtime Modules: -
Build Modules: -

### Reproduction

- Create a new nuxt3 project ( doesn't matter if you are using rc10 )
- Create a new simple component to upload an image 
- Put in the component an input type file like as follows :
`<input type="file" id="file-input" multiple @change="onInputChange" />`
- Add a script tag and create a new method `onInputChange` like this : 
`onInputChange (e) {
  const file = e.target.files
  await $fetch('/api/uploadFile', {
    method: "POST",
   body: {
     files: file
   },
   headers : {
     "Content-Type": "multipart/form-data"
   }
  })
}`
- After that create a new ssr api in the `server/api` directory like this : 

> import FormData from "form-data";
import { manageErrorServer } from "~~/utils/manageErrorServer";
export default defineEventHandler(async (event) => {
    const config = useRuntimeConfig();
    const body = await useBody(event);
    try{
      const res: Record<string, any> = await $fetch(
          ${config.API_BASE_URL}+"smart-product",{ 
            method: 'POST',
            body: body,
            headers: {
               "Content-Type": "mutlipart/form-data"
             }
          }
        )
      return res
    } catch (e) {
       console.log("erreur " + e.response)
       manageErrorServer(e);
    }
  });

### Describe the bug

The multipart/form-data is not sent by $fetch, either on the first call to the ssr api or on the second call to the backend

**Comment by danielroe** [comment]
Let's follow upstream issue on https://github.com/unjs/ohmyfetch/issues/37.

**Comment by huang-julien** [comment]
Hi doesn't this work when setting a `FormData` to the body ? I use the solution provided here https://github.com/nuxt/framework/discussions/4409#discussioncomment-2679474

**Comment by RijaCloud** [comment]
Hi thanks for the answer, @danielroe I tried your solution but it still doesn't work on my side

**Comment by ivanempire** [comment]
Chiming in here - seeing similar issues on my end where the content-type header is not respected (tried using FormData, querystring, regular body object...). If anyone wants to unblock themselves, `axios` works as expected...

**Comment by tak-ho** [comment]
Me too, its not working for me, I raise another issue about posting file fail using FormData with ofetch
https://github.com/unjs/ofetch/issues/186

**Comment by lenimbugua** [comment]
Similar issue, if anyone knows a workaround without needing to install axios please.

**Comment by kurashina-design** [comment]
My workaround:

`yarn add form-data`
`yarn add h3-formidable`

As a server api,

```
import {readFiles} from 'h3-formidable';
import FormData from 'form-data';
import fs from 'fs';
import Media from '~/types/media';

export default eventHandler(async (event) => {

    const formDataKey = 'files[]';

    // Temporary server upload  by using h3-formidable
    const files = await readFiles(event, {uploadDir: 'assets/tmp'});

    // @ts-ignore
    const fileArr = Array.from(files[formDataKey]);

    const formData = new FormData();

    fileArr.forEach(file => formData.append(formDataKey, fs.createReadStream(file.filepath), file.originalFilename!));

    const fetchOptions = {
        // formData's headers are important!
        headers: { ...formData.getHeaders(), 'Authorization': `Bearer ${ACCESS_TOKEN}` }
    } as FetchOptions;
    fetchOptions.method = 'POST';
    fetchOptions.body = formData;

    try {
        // remote upload
        return await $fetch<Media[]>(`${REMOTE_SERVER_UPLOAD_URL}`, fetchOptions);
    } catch (error) {
        someErrorHandling(error);
    } finally {
        // Temporary files should be removed.
        fileArr.forEach(file => fs.rmSync(file.filepath));
    }
})
```

**Comment by RijaCloud** [comment]
Our workaroud for this was to add proxy like the following in `nuxt.config`
````
proxy: {
        options:[ 
            {
                target: process.env.API_HOST_URL,
                changeOrigin: true,
                pathRewrite: {
                    "^/api/media/upload": "/api/v1/upload",
                },
                pathFilter: ["/api/media/upload"],
            },
            {
                target: process.env.API_HOST_URL,
                changeOrigin: true,
                pathRewrite: {
                    "^/api/product/": "/api/v1/product",
                },
                pathFilter: ["/api/product"],
            },
        ]
    }, 
````
